# HG changeset patch
# User Rik <rik@octave.org>
# Date 1475258741 25200
#      Fri Sep 30 11:05:41 2016 -0700
# Branch stable
# Node ID d14c44923705d1faf325754a4ee8841e87f6b7d7
# Parent  3fd1b248bb27bd3d166e02cd79b2722c82a5ae9e
Use fix, rather than round, for "direct" image data which is not an integer.

* graphics.cc (convert_cdata_2): Use octave::math::fix instead of round on
"direct", real data.

diff --git a/libinterp/corefcn/graphics.cc b/libinterp/corefcn/graphics.cc
--- a/libinterp/corefcn/graphics.cc
+++ b/libinterp/corefcn/graphics.cc
@@ -859,17 +859,17 @@ screen_size_pixels (void)
 static void
 convert_cdata_2 (bool is_scaled, bool is_real, double clim_0, double clim_1,
                  const double *cmapv, double x, octave_idx_type lda,
                  octave_idx_type nc, octave_idx_type i, double *av)
 {
   if (is_scaled)
     x = octave::math::fix (nc * (x - clim_0) / (clim_1 - clim_0));
   else if (is_real)
-    x = octave::math::round (x - 1);
+    x = octave::math::fix (x - 1);
 
   if (octave::math::isnan (x))
     {
       av[i]       = x;
       av[i+lda]   = x;
       av[i+2*lda] = x;
     }
   else
